app/controllers/papers_controller.rb

```diff
@@ -5,7 +5,10 @@ class PapersController < ApplicationController
   # GET /papers
   # GET /papers.json
   def index
-    @papers = Paper.all
+    @papers = Paper.select('papers.*,sum(cached_votes_up) as total_votes')
+    .joins(:references)
+    .group('papers.id')
+    .order(('total_votes DESC, ' if params[:sort] != 'pub_date').to_s + 'papers.published_at DESC NULLS LAST')
```

Comment:
That's cool. I'll probably have to rewrite the paper references like that.

On Sat, Sep 10, 2016 at 3:30 AM, wsf246 <notifications@github.com> wrote:

> In app/controllers/papers_controller.rb
> <https://github.com/briankung/projectcredo/pull/70#discussion_r78273837>:
>
> > @@ -5,7 +5,10 @@ class PapersController < ApplicationController
> >    # GET /papers
> >    # GET /papers.json
> >    def index
> > -    @papers = Paper.all
> > +    @papers = Paper.select('papers.*,sum(cached_votes_up) as total_votes')
> > +    .joins(:references)
> > +    .group('papers.id')
> > +    .order(('total_votes DESC, ' if params[:sort] != 'pub_date').to_s + 'papers.published_at DESC NULLS LAST')
>
> sure. Select selects the columns you want. paper.* selects all the columns
> in the papers table and sum(cached_votes_up) sums up all the votes based on
> what you group by. I'm joining the references table to the papers tables
> and grouping by papers.id so it's giving me one paper per result and
> summing all the votes that way, which is all votes on each of the
> references associated with that paper. The order sorts by the total votes
> for that paper and then by publication date.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/briankung/projectcredo/pull/70/files/e5240cef8e29cf5c4de4f689dc3026f25a5f6af5#r78273837>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACtGxydHS11At73TDZ5QuZub0pbzMas7ks5qomq_gaJpZM4J2i6I>
> .
>



-- 
Brian Kung
http://about.me/briankung/
<http://www.facebook.com/brian.kung> <http://www.linkedin.com/in/briankung>
<http://www.twitter.com/briankung>

BE A UNICORN!